backend/src/cmd/import_realm_tree.rs

```diff
@@ -62,31 +64,133 @@ pub(crate) async fn run(args: &Args, config: &Config) -> Result<()> {
         .context("failed to check/run DB migrations")?;
     let conn = db.get().await?;
 
-    let mut dummy_blocks = DummyBlocks::new(args.dummy_blocks, &**conn).await?;
-
+    if args.dummy_blocks {
+        add_dummy_blocks(&mut root, &**conn).await?;
+    }
+    
     info!("Starting to insert realms into the DB...");
-    insert_realm(&**conn, &root, 0, &mut dummy_blocks).await?;
+    insert_realm(&**conn, &root, 0).await?;
     info!("Done inserting realms");
 
     Ok(())
 }
 
+/// Visits all empty realms in the given tree recursively and calls `f` for each realm.
+fn for_all_empty_realms(realm: &mut Realm, mut f: impl FnMut(&mut Realm)) {
+    if realm.blocks.is_empty() {
```

Comment:
This is not the correct position for this check. It needs to be in the inner function around the `f(realm)` call. Otherwise only ever the root realm is checked.